Can't pass arguments to WeakMap constructor via super call in subclass

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  3.0.0-dev.20180609

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** `WeakMap`, `super arguments`

**Code**

```ts
class ExtendedWeakMap extends WeakMap {
    constructor(...args) {
        super(...args);
    }
}
```

Compiled with `tsc --target ES2015 broken.ts`

**Expected behavior:** Output essentially identical to input.

**Actual behavior:** Compiler error:

```
broken.ts:3:9 - error TS2556: Expected 0 arguments, but got 1 or more.

3         super(...args);
          ~~~~~~~~~~~~~~
```

**Playground Link:** [here](https://www.typescriptlang.org/play/#src=class%20ExtendedWeakMap%20extends%20WeakMap%20%7B%0D%0A%20%20%20%20constructor(...args)%20%7B%0D%0A%20%20%20%20%20%20%20%20super(...args)%3B%0D%0A%20%20%20%20%7D%0D%0A%7D)

**Related Issues:** This is not #23551; that issue does not involve args to super call.

**Notes:**

Having examined the declaration in `src/lib/es2015.collection.d.ts`:

```ts
interface WeakMapConstructor {
    new (): WeakMap<object, any>;
    new <K extends object, V>(entries?: ReadonlyArray<[K, V]> | null): WeakMap<K, V>;
    readonly prototype: WeakMap<object, any>;
}
declare var WeakMap: WeakMapConstructor;
```

…and noting that this code compiles fine:

```ts
var o = {};
var w = new WeakMap([[o, 42]]);
```

…I suspect the error would go away if I added an appropriate type declaration to `ExtendedWeakMap`. 
 But is it not the case that every valid JS program is supposed to be as valid TS program?

--- a/src/lib/es2015.collection.d.ts
+++ b/src/lib/es2015.collection.d.ts
@@ -30,8 +30,7 @@ interface WeakMap<K extends object, V> {
 }
 
 interface WeakMapConstructor {
-    new (): WeakMap<object, any>;
-    new <K extends object, V>(entries?: ReadonlyArray<[K, V]> | null): WeakMap<K, V>;
+    new <K extends object = object, V = any>(entries?: ReadonlyArray<[K, V]> | null): WeakMap<K, V>;
     readonly prototype: WeakMap<object, any>;
 }
 declare var WeakMap: WeakMapConstructor;

